src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccount.fs

```diff
@@ -0,0 +1,76 @@
+﻿module Deciders.BankAccount
+
+open System
+
+type BankAccountOpened =
+    { BankAccountId: Guid
+      AccountNumber: string
+      ClientId: Guid
+      CurrencyISOCode: string
+      CreatedAt: DateTimeOffset
+      Version: int64 }
+
+type DepositRecorded =
+    { BankAccountId: Guid
+      Amount: decimal
+      CashierId: Guid
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type CashWithdrawnFromATM =
+    { BankAccountId: Guid
+      Amount: decimal
+      ATMId: Guid
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type BankAccountClosed =
+    { BankAccountId: Guid
+      Reason: string
+      ClosedAt: DateTimeOffset
+      Version: int64 }
+
+type Event =
+    | BankAccountOpened of BankAccountOpened
+    | DepositRecorded of DepositRecorded
+    | CashWithdrawnFromATM of CashWithdrawnFromATM
+    | BankAccountClosed of BankAccountClosed
+
+type BankAccountStatus =
+    | Opened
+    | Closed
+
+type BankAccount =
+    { Id: Guid
+      Status: BankAccountStatus
+      Balance: decimal
+      Version: int64 }
+
+    static member Create(opened: BankAccountOpened) =
+        { Id = opened.BankAccountId
+          Status = Opened
+          Balance = 0M
+          Version = 0 }
+
+    member this.Apply(depositRecorded: DepositRecorded) =
+        { this with
+            Balance = this.Balance + depositRecorded.Amount
+            Version = depositRecorded.Version }
+
+    member this.Apply(depositRecorded: CashWithdrawnFromATM) =
+        { this with
+            Balance = this.Balance - depositRecorded.Amount
+            Version = depositRecorded.Version }
+
+    member this.Apply(depositRecorded: BankAccountClosed) =
+        { this with
+            Status = BankAccountStatus.Closed
+            Version = depositRecorded.Version }
+
```

Comment:
Small tip, instead of having overloads of the `Apply` method you could also have a single `Apply` which takes the `Event` and then do a `match e with` expression. The benefit of that over the overloads is that the F# compiler will give you a warning if you forget to cover a case (e.g. if you add a new event later on). With overloads there is no compile time check to ensure that your bank account can actually apply all possible events.

Reply:
Yes, this is what the evolve function does in a decider. Apply is not necessary and gives a `mutable` feeling to it.

Reply:
Eventually, I follow your advice and decided to use Marten conventions alone and follow your suggestions. This PR is not related to Marten, so it's better not to spoil it with our quirks. I updated in https://github.com/oskardudycz/Ogooreck/pull/13/commits/8537bcf8349ecc486a5c426c27e12632a26cf7fc.